### PR TITLE
Fix maximum mipmap level

### PR DESCRIPTION
**Pull request by vickenty**
Fixes maximum mipmap level passed to ``set_mipmap_range()`` to be one less than number of mipmaps and adds an example to show that things still work correctly after that.

### CODE REVIEW

File: src/backend/gl/src/tex.rs [hunk 1]
```diff
@@ -384,7 +384,7 @@ pub fn make_without_storage(gl: &gl::Gl, info: &TextureInfo) ->
         _ => return Err(TextureError::UnsupportedSampling),
     }
 
-    set_mipmap_range(gl, target, (0, info.levels));
+    set_mipmap_range(gl, target, (0, info.levels - 1));
```
Comment (kvark) [inline src/backend/gl/src/tex.rs]: Do we catch `info.levels == 0` case anywhere?

File: examples/mipmap/main.rs [hunk 2]
```diff
@@ -0,0 +1,176 @@
+// Copyright 2014 The Gfx-rs Developers.
```
Comment (kvark) [inline examples/mipmap/main.rs]: 2015 ;)

**Comment by kvark** [comment]
Thanks @vickenty , looks very clean!
Just a couple of minor points to address.

File: src/backend/gl/src/tex.rs (hunk 1 above)
Comment (vickenty) [inline src/backend/gl/src/tex.rs]: In [Factory.create_texture()](https://github.com/gfx-rs/gfx/blob/master/src/backend/gl/src/factory.rs#L267)

**Comment by kvark** [comment]
@homu r+

**Pull request merged**